sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -1261,3 +1261,51 @@ def test_gaussian_mixture_setting_best_params():
         "lower_bound_",
     ]:
         assert hasattr(gmm, attr)
+
+
+def test_init_param():
+    # Check that the init_param options produces valid output.
+    # Compare all to default - kmeans.
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng)
+
+    n_components = rand_data.n_components
+    X = rand_data.X['full']
+    gmm = GaussianMixture(n_components=n_components, random_state=rng,
+                          init_params='kmeans')
+    gmm.fit(X)
+    default_means = np.sort(gmm.means_.flatten())
+
+    INIT_TYPE = ['random', 'rand_data', 'k-means++', 'kmeans']
+
+    for init in INIT_TYPE:
+        gmm = GaussianMixture(n_components=n_components, random_state=rng,
+                              init_params=init, tol=1e-06, max_iter=1000)
+        gmm.fit(X)
+        assert_almost_equal(default_means, np.sort(gmm.means_.flatten()),
+                            decimal=1)
```

Comment:
I don't find the `flatten` + `sort` combo convincing. I would rather compute the value of the value of the loglikelihood of the training set and compare that (`gmm.score(X)`).

Or alternatively, combine the pairwise cosine similarities of the means and check that the similarity matrix has one and only one entry with with a value larger than 0.99 in each column and the other values are below 0.01.

Side note: nowadays, we tend to use `assert_allclost` with the `atol` and `rtol` parameters to control for the precision is a more maintainable way (instead of the `decimal` parameter of `assert_almost_equal`).

Reply:
Thank you @ogrisel , I had a busy week but I will look into all of your suggestions ASAP :)

Reply:
Hi @ogrisel, I am attempting to implement the pairwise cosine similarity solution. The first condition works well, however, I don't seem to be getting values below 0.01 for the rest of the entries.

Here's a snippet of the proposed solution:

```
pw_sim = cosine_similarity(default_means, gmm.means_) #shape of both means: n_components, n_features
assert np.all(np.sum(pw_sim > 0.99, axis=0) == 1) and np.all(np.sum(pw_sim <= 0.01, axis=0) == pw_sim.shape[0]-1)

```
The test output indicates that most of the time the non >0.99 values have values greater than 0.01. Does that make sense to you?